### PR TITLE
Remove Basic_auth from openapi because the backend does not support it

### PR DESCRIPTION
**Pull request by bzwei**
It actually causes trouble in the client used by QE.

### CODE REVIEW

**Comment by bzwei** [comment]
@gmcculloug can you merge it? QE is waiting for it.

**Comment by mkanoor** [comment]
@bzwei How would this work for external facing, customers who are using the RHN login which uses userid/password? That is basic auth inside of HTTPS

**Comment by mkanoor** [comment]
@bzwei At least from the catalog perspective we use the basic auth so that customers using the API externally can access it.
https://github.com/mkanoor/catalog-api-client-ruby/blob/b476fbe6a9695f0206be79cce8eff2f527dd78d9/lib/catalog-api-client/configuration.rb#L192

**Comment by bsquizz** [comment]
Aren't all the apps in the public-facing environments accessed via 3scale, with 3scale handling the auth and passing on an `x-rh-identity` header? Your app is not directly receiving a Basic authorization header is it?

**Comment by syncrou** [comment]
@bsquizz once on any of the environments yes.  But we have been using basic auth when testing one service external to 3scale that requires another service behind 3scale to complete that test.

**Comment by bsquizz** [comment]
Does disabling basic auth in the spec affect how the app works? It will still accept basic auth even if the spec changes, or are you generating your app code from the spec?

**Comment by bzwei** [comment]
@mkanoor @syncrou Our backend does not validate username/password, how does this requirement in openapi help? Zvika is facing difficulties currently when doing smoking tests. I guess Beni will have similar problem too.

**Comment by syncrou** [comment]
@bsquizz we're not generating app code from the spec. I just wanted to make sure that we could still use basic auth whether it existed in the spec or not. Thanks.

File: spec/controllers/v1.0/actions_controller_spec.rb [hunk 1]
```diff
@@ -66,8 +66,8 @@
     end
 
     context 'when request is actionable' do
-      let!(:stage1) { create(:stage, :id => "1", :state => Stage::NOTIFIED_STATE, :request => req, :tenant_id => tenant.id) }
-      let!(:stage2) { create(:stage, :id => "2", :state => Stage::PENDING_STATE, :request => req, :tenant_id => tenant.id) }
```
Comment (syncrou) [inline spec/controllers/v1.0/actions_controller_spec.rb]: Were these changes needed as part of the basic auth change?

Comment (bzwei) [reply spec/controllers/v1.0/actions_controller_spec.rb]: no. it failed the build. was introduced by a previous PR.

**Comment by zsadeh** [comment]
@gmcculloug 
Well I guess this PR should be closed as we decided not to remove security settings from spec file.
I guess we need a new ticket to add all supported security scheme to the spec file of approval and catalog.

**Comment by gmcculloug** [comment]
@zsadeh Agree to close.  Please open a git issue for the followup work.  Thanks.
